actionpack/lib/action_dispatch/middleware/server_timing.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module ActionDispatch
+  class ServerTiming
+    class << self
+      def add_timing(key:, dur: nil, desc: nil)
+        timings[key] = { dur: dur, desc: desc }
+      end
+
+      def to_header
+        headers = []
+        timings.each_pair do |key, data|
+          header = "#{key}"
+          header += ";desc=\"#{data[:desc]}\"" if data[:desc]
+          header += ";dur=#{data[:dur]}" if data[:dur]
+          headers << header
+        end
+        headers.join(", ")
+      end
+
+      def measure(key:, desc: nil, &blk)
```

Comment:
Where is this going to be called from ?

Reply:
This is a convenience method for application code to call. And the intended usage is something like this: 
``` ruby
class MyController < ApplicationController
  def index
    ActionDispatch::ServerTiming.measure(key: "read_thing", desc: "reading") do
      @things = Thing.all
    end
  end
end
```

The idea is to avoid having to avoid repeatedly manually instrument code through: `ActiveSupport::Notifications.instrument("measure.server_timing", desc: desc, key: key) do`

Reply:
I feel like `ActionDispatch::ServerTiming.measure("read_thing", desc: "reading")` might be a better API, WDYT?

Reply:
good call, I personally prefer `kwarg` arguments on 2+ arguments but since the `desc` is optional I can see this mostly being used as: `ActionDispatch::ServerTiming.measure(key: "read_thing")` and the `key` there feels redundant.